contentcuration/contentcuration/views/internal.py

```diff
@@ -590,6 +590,11 @@ def convert_data_to_nodes(user, content_data, parent_node):
                             user, new_node, slides, node_data["files"]
                         )
 
+                    new_node.mark_complete()
+
+                    if not new_node.complete:
+                        new_node.save()
```

Comment:
I'm wondering if it would be clearer to move this new code up right after `new_node` creation, to show that this has no effect on the state of the node besides the `complete` status. At least, that's what I see to be the case, jumping into the functions called in between. So I think moving it up would save someone having to do the same to see it clearly when looking at this code in its complete, non-diff, form.

Also, "how we should audit this" -- maybe we can add some logging to this to get an idea how often it's occurring? We could hold off on saving until we get some data.

Reply:
> Also, "how we should audit this" -- maybe we can add some logging to this to get an idea how often it's occurring? We could hold off on saving until we get some data.

Yeah - maybe we could do the same trick we do with the `SlowSyncError` and just raise an error that we immediately catch to report to Sentry? Then we can just keep an eye on the number of sentry events for this to see whether it's an issue or not.

Reply:
Yeah, that would work well!

Reply:
> I'm wondering if it would be clearer to move this new code up right after new_node creation, to show that this has no effect on the state of the node besides the complete status. At least, that's what I see to be the case, jumping into the functions called in between. So I think moving it up would save someone having to do the same to see it clearly when looking at this code in its complete, non-diff, form.

This made sense to me - but when I went to double check, the reason I put it down here was because for non-topic/exercise nodes, it makes assertions about the associated files, so it has to happen after the file mapping steps that are happening in between. I'll add a code comment to clarify instead.

Reply:
Ah yeah okay, that makes sense.